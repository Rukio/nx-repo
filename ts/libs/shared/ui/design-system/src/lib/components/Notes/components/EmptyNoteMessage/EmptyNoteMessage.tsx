import { Box, makeSxStyles, StickyNote2Icon, Typography } from '../../../..';
import { NOTES_TEST_IDS } from '../../testIds';

const makeStyles = () =>
  makeSxStyles({
    root: (theme) => ({
      display: 'flex',
      flexDirection: 'column',
      alignItems: 'center',
      px: theme.spacing(2),
      color: theme.palette.text.secondary,
    }),
  });

const EmptyNoteMessage = () => {
  const styles = makeStyles();

  return (
    <Box sx={styles.root} data-testid={NOTES_TEST_IDS.EMPTY_MESSAGE}>
      <StickyNote2Icon />
      <Typography mt={1} variant="subtitle2">
        No notes have been added
      </Typography>
    </Box>
  );
};

export default EmptyNoteMessage;
