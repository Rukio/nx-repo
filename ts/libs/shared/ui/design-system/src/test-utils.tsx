import React from 'react';
import { render, RenderOptions } from '@testing-library/react';
import { ThemeProvider, theme } from './lib';
import type { Config as UserEventSetupConfig } from '@testing-library/user-event/setup/setup';
import userEvent from '@testing-library/user-event';

type CustomRenderOptions = {
  renderOptions?: Omit<RenderOptions, 'wrapper'>;
  userEventOptions?: UserEventSetupConfig;
};

const WrappedProviders: React.FC<{ children: React.ReactNode }> = ({
  children,
}) => {
  return <ThemeProvider theme={theme}>{children}</ThemeProvider>;
};

const customRender = (
  ui: React.ReactElement,
  options?: CustomRenderOptions
) => {
  return {
    user: userEvent.setup(options?.userEventOptions),
    ...render(ui, { wrapper: WrappedProviders, ...options?.renderOptions }),
  };
};

export * from '@testing-library/react';
export { customRender as render, userEvent };
