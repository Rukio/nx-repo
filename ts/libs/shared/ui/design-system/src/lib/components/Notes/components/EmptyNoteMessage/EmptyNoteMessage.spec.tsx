import { render } from '../../../../../test-utils';
import EmptyNoteMessage from './EmptyNoteMessage';

describe('EmptyNoteMessage', () => {
  test('should render snapshot correctly', () => {
    const { asFragment } = render(<EmptyNoteMessage />);
    expect(asFragment()).toMatchSnapshot();
  });
});
