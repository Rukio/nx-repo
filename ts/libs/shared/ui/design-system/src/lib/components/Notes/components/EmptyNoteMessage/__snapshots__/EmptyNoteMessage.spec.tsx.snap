// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EmptyNoteMessage should render snapshot correctly 1`] = `
<DocumentFragment>
  .emotion-0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  padding-left: 16px;
  padding-right: 16px;
  color: #5E696F;
}

.emotion-1 {
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  width: 1em;
  height: 1em;
  display: inline-block;
  fill: currentColor;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  -webkit-transition: fill 200ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  transition: fill 200ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  font-size: 1.5rem;
}

.emotion-2 {
  margin: 0;
  font-weight: 600;
  font-size: 0.875rem;
  line-height: 1.428;
  letter-spacing: 0.016rem;
  font-family: Open Sans,Segoe UI,Tahoma,sans-serif;
  margin-top: 8px;
}

<div
    class="MuiBox-root emotion-0"
    data-testid="notes-empty-message"
  >
    <svg
      aria-hidden="true"
      class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium emotion-1"
      data-testid="StickyNote2Icon"
      focusable="false"
      viewBox="0 0 24 24"
    >
      <path
        d="M19 3H4.99C3.89 3 3 3.9 3 5l.01 14c0 1.1.89 2 1.99 2h10l6-6V5c0-1.1-.9-2-2-2zM7 8h10v2H7V8zm5 6H7v-2h5v2zm2 5.5V14h5.5L14 19.5z"
      />
    </svg>
    <h6
      class="MuiTypography-root MuiTypography-subtitle2 emotion-2"
    >
      No notes have been added
    </h6>
  </div>
</DocumentFragment>
`;
